src/workloads/query/TelemetryQueryShapes.yml

```diff
@@ -0,0 +1,129 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload tests performance of telemetry with different query shapes.
+
+KeyWords:
+- telemetry
+- query
+- shape
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 400
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection Collection0
+  DocumentCount: &DocumentCount 1e6
+  Repeat: &Repeat 1000
+  Threads: &Threads 32
+  MaxPhases: &MaxPhases 6
+  nop: &Nop {Nop: true}
+  dbname: &db test
+
+ActorTemplates:
+- TemplateName: ComputeFindQuery
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "ComputeFindQuery"}}
+    Type: CrudActor
+    Threads: {^Parameter: {Name: "Threads", Default: 1}}
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "OnlyActiveInPhase", Default: 1024}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: *Repeat
+          Database: *Database
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter: {^Parameter: {Name: "Filter", Default: {a: 1, b: 2, x0: {$exists: false}}}}
+
+Actors:
+# Clear any pre-existing collection state.
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 4
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document:
+          a1: &integer {^RandomInt: {min: -100, max: 100}}
+          a2: *integer
+        Indexes:
+        - keys: {a1: 1, a2: 1}
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+
+# 32 threads with distinct query shapes
+- ActorFromTemplate:
+    TemplateName: ComputeFindQuery
+    TemplateParameters:
+      Name: "AlternatingShapes"
+      Threads: 32
+      Filter:
+        {$and:
+           [{^Object: {
+             withNEntries: 3,
```

Comment:
This is cool, but I'm curious if you could do
```
{$and: [
  {a1: 1},
  {a2: 1},
  {^Object: {...}}
]}
```

That way it would be more obvious that we're filtering primarily by "a1" and "a2" and the extra thing is the source of randomness to change the query shape.

Is that more or less the intention here? Or did you want some queries to not have "a1" or "a2" for some reason?

Reply:
> @cswanson310 I changed the workload to have 2 actors, one which oscillates between query shapes on 32 threads, and one that runs the same shape on 32 threads. It actually seems like oscillating shapes had worse performance, which is different from my initial analysis with the past workload. I was thinking maybe the extra time taken could be due to the building of the object / creating the random field name, but I ran the workload with the one shape query going through the same process and am still getting the same results. Here is the link to the new sheet with the results: https://docs.google.com/spreadsheets/d/1_cLQ5sZ3A-4ag05b46n1GZAKSTTzm_YzUqeBgSXFCOg/edit#gid=0

This is quite interesting! I am just starting to look through, but it might be that the way you constructed the query here will sometimes skip the "a1" or "a2" predicates which are the indexable predicates?

Or is the `determenistic: true` part mean that it will always walk in order? If that's so, I might still recommend changing to the different $and for clarity, or at least adding a comment making this clear.

Reply:
The `deterministic: true` part means that it will go in order, so we always had "a1" and "a2", but I agree it is cleaner to have them separately. I think for some reason I thought I couldn't do it that way, but I just tried again and it works!